src/table/operations.jl

```diff
@@ -1,63 +1,127 @@
+        if isvalid(_chunk)
+            m = TableOperations.map(_f, _chunk)
+            Tables.materializer(_chunk)(m)
+        else
+            _chunk
+        end
     end
-    result_vector = map(c -> Dagger.spawn(chunk_wrap, c, f), d.dtable.chunks)
-    Dagger.@spawn vcat(result_vector...)
+    chunks = map(c -> Dagger.spawn(chunk_wrap, c, f), d.chunks)
+    DTable(chunks, d.tabletype)
 end
 
 """
-    reduce(f_reduce::Function, f_row::Function, d::DTableRows; init)
+    reduce(f, d::DTable; cols=nothing, [init]) -> NamedTuple
+
+Reduces `d` using function `f` applied on all columns of the DTable.
+
+By providing the kwarg `cols` as a `Vector{Symbol}` object it's possible 
+to restrict the reduction to the specified columns.
+The reduced values are provided in a NamedTuple under names of reduced columns.
 
-Reduces `d` using function `f_reduce` on values obtained from applying `f_row` on its rows.
+For the `init` kwarg please refer to `Base.reduce` documentation,
+as it follows the same principles. 
 
 # Examples
 ```
-    d = DTable((a=rand(1000), b=rand(1000)), chunksize=100)
-    r = reduce(+, row -> row.a + row.b, eachrow(d); init=0.0)
-    fetch(r)
+julia> d = DTable((a = [1, 2, 3], b = [1, 1, 1]), 2);
+
+julia> r1 = reduce(+, d)
+EagerThunk (running)
+
+julia> fetch(r1)
+(a = 6, b = 3)
+
+julia> r2 = reduce(*, d, cols=[:a])
+EagerThunk (running)
+
+julia> fetch(r2)
+(a = 6,)
 ```
 """
-function reduce(f_reduce::Function, f_row::Function, d::DTableRows; init)
-    chunk_results = map(c -> Dagger.spawn(_chunk_reduce, f_reduce, f_row, c, init), d.dtable.chunks)
-    reduce_chunks = (f, _init, c_res...) -> reduce(f, c_res; init=_init)
-    Dagger.@spawn reduce_chunks(f_reduce, init, chunk_results...)
-end
+function reduce(f, d::DTable; cols=nothing::Union{Nothing, Vector{Symbol}}, init=Base._InitialValue())
+    # TODO replace this with checking the colnames in schema, once schema handling gets introduced
+    if length(d.chunks) > 0
+        columns = isnothing(cols) ? Tables.columnnames(Tables.columns(_retrieve(d.chunks[1]))) : cols
+    else
+        return Dagger.@spawn NamedTuple()
+    end
 
-function _chunk_reduce(f_reduce, f_row, chunk, init)
-    mapped = chunk |> TableOperations.map(x -> (result = f_row(x)))
-    reduce(f_reduce, mapped; init=init)
+    chunk_reduce = (_f, _chunk, _cols, _init) -> begin
+        values = [reduce(_f, Tables.getcolumn(_chunk, c); init=deepcopy(_init)) for c in _cols]
+        (; zip(_cols, values)...)
+    end
+    chunk_reduce_results = [Dagger.@spawn chunk_reduce(f, c, columns, deepcopy(init))  for c in d.chunks]
+
+    construct_single_column = (_col, _chunk_results...) -> getindex.(_chunk_results, _col)
+    result_columns = [Dagger.@spawn construct_single_column(c, chunk_reduce_results...) for c in columns]
+
+    reduce_result_column = (_f, _c, _init) -> reduce(_f, _c; init=_init)
+    reduce_chunks = [Dagger.@spawn reduce_result_column(f, c, deepcopy(init)) for c in result_columns]
```

Comment:
So this part first takes the per chunk results and makes columns out of them and then reduces them.
I tried treereduce instead of this and it was noticeably slower.
I think for the DTable where we know there's not going to be more than <reasonable number> of chunks this could potentially be always faster than treereduce
Is there any case to use treereduce instead? Actually multimachine distributed maybe?